x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -702,7 +702,7 @@ protected Clock getClock() {
 
             try (XContentBuilder stateMapping = ElasticsearchMappings.stateMapping()) {
                 IndexTemplateMetaData stateTemplate = IndexTemplateMetaData.builder(AnomalyDetectorsIndex.jobStateIndexName())
-                        .patterns(Collections.singletonList(AnomalyDetectorsIndex.jobStateIndexName()))
+                        .patterns(Collections.singletonList(AnomalyDetectorsIndex.jobStateIndexName() + "*"))
```

Comment:
Would it be better to rename `AnomalyDetectorsIndex.jobStateIndexName()` to `AnomalyDetectorsIndex.jobStateIndexPattern()`, returning `".ml-state*"` and always use that when matching state indices?

Reply:
Potentially, I can go ahead and do that in this PR, I was worried about simply making sure that if it was an alias backed by more than one index, that we would OK.

Reply:
I kept the old name and added a new one. Many places that write to the index and such should probably hit the index directly (sans wildcard) for now until we add the aliases.